sql/api/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1162,7 +1162,16 @@ object functions {
    * @group agg_funcs
    * @since 4.0.0
    */
-  def listagg(e: Column, delimiter: Column): Column = Column.fn("listagg", e, delimiter)
+  def listagg(e: Column, delimiter: String): Column = Column.fn("listagg", e, lit(delimiter))
```

Comment:
I think it is good to have `def listagg(e: Column, delimiter: String)`, but I also want to keep `def listagg(e: Column, delimiter: Column)`